multinode/transaction_sender_test.go

```diff
@@ -138,6 +139,26 @@ func TestTransactionSender_SendTransaction(t *testing.T) {
 		require.EqualError(t, err, "context canceled")
 	})
 
+	t.Run("Context cancelled while sending results does not cause invariant violation", func(t *testing.T) {
+		requestContext, cancel := context.WithCancel(tests.Context(t))
+		mainNode := newNode(t, nil, func(_ mock.Arguments) {
+			cancel()
+		})
+
+		lggr, observedLogs := logger.TestObserved(t, zap.WarnLevel)
+
+		_, txSender := newTestTransactionSender(t, RandomID(), lggr,
+			[]Node[ID, TestSendTxRPCClient]{mainNode}, nil)
+
+		_, _, err := txSender.SendTransaction(requestContext, nil)
+		require.EqualError(t, err, "context canceled")
+
+		// Insure no invariant violation occurred
+		require.Never(t, func() bool {
```

Comment:
We should not use never as it's forces test to run until timeout to pass. In case of CI it might be 10 minutes.